src/main/java/org/hypertrace/gradle/dependency/HypertraceDependencySettingsPlugin.java

```diff
@@ -0,0 +1,208 @@
+                              this.addProjectExtension(project);
+                          this.addBomDependencyIfRequested(
+                              project, settingExtension, projectExtension);
+                          if (settingExtension.useDependencyLocking.get()) {
+                            this.addDependencyLocking(project, projectExtension);
+                            this.addLockTask(project, projectExtension);
+                          }
+                        }));
+  }
+
+  private void addBomDependencyIfRequested(
+      Project project,
+      DependencyPluginSettingExtension settingExtension,
+      DependencyPluginProjectExtension projectExtension) {
+    project.afterEvaluate(
+        unused -> {
+          projectExtension.autoApplyBom.finalizeValueOnRead();
+          if (projectExtension.autoApplyBom.getOrElse(false)) {
+            project
+                .getDependencies()
+                .addProvider(
+                    resolveBomHostConfigurationName(project),
+                    project
+                        .getDependencies()
+                        .platform(resolveBomDependency(project, settingExtension)));
+          }
+        });
+  }
+
+  private String resolveBomHostConfigurationName(Project project) {
+    return Optional.ofNullable(
+            project.getConfigurations().findByName(JavaPlugin.API_CONFIGURATION_NAME))
+        .orElseGet(
+            () ->
+                project.getConfigurations().getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME))
+        .getName();
+  }
+
+  private Provider<MinimalExternalModuleDependency> resolveBomDependency(
+      Project project, DependencyPluginSettingExtension settingExtension) {
+    return settingExtension
+        .catalogName
+        .map(name -> getCatalogByName(project, name))
+        .flatMap(
+            catalog ->
+                settingExtension.bomArtifactName.flatMap(
+                    artifactName ->
+                        catalog
+                            .findLibrary(artifactName)
+                            .orElseThrow(
+                                () ->
+                                    new GradleException(
+                                        "No artifact available in catalog declared alias "
+                                            + artifactName))));
+  }
+
+  private DependencyPluginSettingExtension createSettingsExtension(Settings target) {
+    return target
+        .getExtensions()
+        .create(
+            DependencyPluginSettingExtension.EXTENSION_NAME,
+            DependencyPluginSettingExtension.class);
+  }
+
+  private DependencyPluginProjectExtension addProjectExtension(Project project) {
+    return project
+        .getExtensions()
+        .create(
+            DependencyPluginProjectExtension.EXTENSION_NAME,
+            DependencyPluginProjectExtension.class);
+  }
+
+  private void addVersionCatalog(
```

Comment:
Will help a great deal if you can add a comment with the snippet of the changes this is automatically making to the settings and build.gradle files.

Reply:
I'm assuming this is similar to adding something like this manually in the settings file
```
versionCatalogs {
        create("commonLibs") {

```

Reply:
Sure, let me go back and add some comments through here. (and yes it is, those specifics lines map to 124-125).